stacks-signer/src/stacks_client.rs

```diff
@@ -91,6 +94,28 @@ impl StacksClient {
         }
     }
 
+    /// Retrieve the current DKG aggregate public key
+    pub fn get_aggregate_public_key(&self) -> Result<Option<Point>, RPCError> {
+        let _burn_block_height = self.get_burn_block_height()?;
+        todo!("Make the read only contract call to retrieve the aggregate public key for the given block height")
+    }
+
+    /// Retreive the DKG aggregate public key vote cast by the signer
+    pub fn get_aggregate_public_key_vote(&self) -> Result<Option<Point>, RPCError> {
+        let _burn_block_height = self.get_burn_block_height()?;
+        todo!("Make the read only contract call to retrieve the aggregate public key vote cast by the sender for a given block height")
+    }
+
+    /// Retreive the current burn block height
+    fn get_burn_block_height(&self) -> Result<u64, RPCError> {
+        todo!("Get the current burn block height from the stacks node")
+    }
+
+    /// Cast the DKG aggregate public key vote
+    pub fn cast_aggregate_public_key_vote(&self, _vote: Point) -> Result<(), RPCError> {
```

Comment:
Clarity contract should expect a public key and verify that the sender is a valid signer who has NOT yet cast their vote.